src/core/request.js

```diff
@@ -59,31 +59,31 @@ Request.prototype.request = function(servers, callback_) {
   this.on('error', function() {});
   this.emit('request', this.remote);
 
-  function doRequest() {
-    if (Array.isArray(servers)) {
-      servers.forEach(function(server) {
-        self.setServer(server);
-        self.remote.request(self);
-      }, self);
-    } else {
-      self.remote.request(self);
+  const timeout = setTimeout(() => {
+    if (typeof callback === 'function') {
+      callback(new RippleError('tejTimeout'));
     }
-  }
 
-  function onReconnect() {
```

Comment:
So I think there's no need to remove the reconnect logic for non-transaction requests; it would be convenient to keep it for read-only requests.